ikarus/finiteelements/mechanics/materials/hyperelastic/concepts.hh

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file concepts.hh
+ * \brief Header file including concepts for hyperelastic material models.
+ * \ingroup  materials
+ */
+
+#pragma once
+
+#include <concepts>
+#include <string>
+
+namespace Ikarus::Concepts {
+
+/**
+ * \concept DeviatoricFunction
+ * \brief Concept to check if the underlying function is a deviatoric function.
+ * \tparam Type of the deviatoric function.
+ *
+ */
+template <typename DF>
+concept DeviatoricFunction = requires(DF dm, const typename DF::PrincipalStretches& lambda) {
+  typename DF::ScalarType;
+  typename DF::PrincipalStretches;
+
+  typename DF::FirstDerivative;
+  typename DF::SecondDerivative;
+  typename DF::MaterialParameters;
+
+  { dm.storedEnergyImpl(lambda) } -> std::same_as<typename DF::ScalarType>;
+  { dm.firstDerivativeImpl(lambda) } -> std::same_as<typename DF::FirstDerivative>;
+  { dm.secondDerivativeImpl(lambda) } -> std::same_as<typename DF::SecondDerivative>;
```

Comment:
The DeviatoricFunction concept and VolumetricFunction are almost the same. I think it would be better to merge these into a concept `Materialfunction`. Please think about a better name!

Then additionally you can add a